pkg/miniogw/gateway-storj.go

```diff
@@ -5,31 +5,143 @@ package storj
 		HideHelpCommand: true,
 	})
 }
 
+// getbucketscallback returns the buckets list
+func (s *storjObjects) getBuckets() (buckets []minio.BucketInfo, err error) {
+	buckets = make([]minio.BucketInfo, len(s.storj.bucketlist))
+	for i, bi := range s.storj.bucketlist {
+		buckets[i] = minio.BucketInfo{
+			Name:    bi.bucket.Name,
+			Created: bi.bucket.Created,
+		}
+	}
+	return buckets, nil
+}
+
+// putobjectcallback function handles to add the uploaded file to the bucket's file list structure
+func (s *storjObjects) uploadFile(bucket, object string, filesize int64, metadata map[string]string) (result minio.ListObjectsInfo, err error) {
+	var fl []minio.ObjectInfo
+	for i, v := range s.storj.bucketlist {
+		// bucket string comparision
+		if v.bucket.Name == bucket {
+			/* append the file to the filelist */
+			s.storj.bucketlist[i].filelist.file.Objects = append(
+				s.storj.bucketlist[i].filelist.file.Objects,
+				minio.ObjectInfo{
+					Bucket:      bucket,
+					Name:        object,
+					ModTime:     time.Now(),
+					Size:        filesize,
+					IsDir:       false,
+					ContentType: "application/octet-stream",
+				},
+			)
+			/* populate the filelist */
+			f := make([]minio.ObjectInfo, len(s.storj.bucketlist[i].filelist.file.Objects))
+			for j, fi := range s.storj.bucketlist[i].filelist.file.Objects {
+				f[j] = minio.ObjectInfo{
+					Bucket:      v.bucket.Name,
+					Name:        fi.Name,
+					ModTime:     fi.ModTime,
+					Size:        fi.Size,
+					IsDir:       fi.IsDir,
+					ContentType: fi.ContentType,
+				}
+			}
+			fl = f
+			break
+		}
+	}
+	result = minio.ListObjectsInfo{
+		IsTruncated: false,
+		Objects:     fl,
+	}
+	return result, nil
+}
+
+// listfilescallback returns the files list for a bucket
+func (s *storjObjects) getFiles(bucket string) (result minio.ListObjectsInfo, err error) {
+	var fl []minio.ObjectInfo
+	for i, v := range s.storj.bucketlist {
+		if v.bucket.Name == bucket {
+			/* populate the filelist */
+			f := make([]minio.ObjectInfo, len(s.storj.bucketlist[i].filelist.file.Objects))
+			for j, fi := range s.storj.bucketlist[i].filelist.file.Objects {
+				f[j] = minio.ObjectInfo{
+					Bucket:      v.bucket.Name,
+					Name:        fi.Name,
+					ModTime:     fi.ModTime,
+					Size:        fi.Size,
+					IsDir:       fi.IsDir,
+					ContentType: fi.ContentType,
+				}
+			}
+			fl = f
+			break
+		}
+	}
+	result = minio.ListObjectsInfo{
+		IsTruncated: false,
+		Objects:     fl,
+	}
+	return result, nil
+}
+
 func storjGatewayMain(ctx *cli.Context) {
-	cmd.StartGateway(ctx, &Storj{})
+	s := &Storj{}
+	s.createSampleBucketList()
+	minio.StartGateway(ctx, s)
+}
+
+//S3BucketList structure
+type S3BucketList struct {
+	bucket   minio.BucketInfo
+	filelist S3FileList
+}
+
+//S3FileList structure
+type S3FileList struct {
+	file minio.ListObjectsInfo
 }
 
 // Storj is the implementation of a minio cmd.Gateway
-type Storj struct{}
+type Storj struct {
+	bucketlist []S3BucketList
```

Comment:
Why not just `buckets` instead of `bucketlist`? The shorter, the better.

Reply:
it holds the list of buckets... thats how I visioned. No change will be made